utils.py

```diff
@@ -481,34 +506,39 @@ def make_skel(app_image_filename):
 
     logging.info("skel64 done!")
 
-def check_commands(commands):
-    missing_cmd = []
-    for cmd in commands:
-        if have_dep(cmd):
-            logging.info(f"* Command {cmd} is installed!")
-        else:
-            logging.warning(f"* Command {cmd} not installed!")
-            missing_cmd.append(cmd)
-
-    if missing_cmd:
-        if config.PACKAGE_MANAGER_COMMAND:
-            message = f"Your {config.OS_NAME} install is missing the command(s): {missing_cmd}. To continue, the script will attempt to install the package(s): {config.PACKAGES} by using ({config.PACKAGE_MANAGER_COMMAND}). Proceed?"
-            if config.OS_NAME == "Steam":
-                subprocess.run([config.SUPERUSER_COMMAND, "steamos-readonly", "disable"], check=True)
-                subprocess.run([config.SUPERUSER_COMMAND, "pacman-key", "--init"], check=True)
-                subprocess.run([config.SUPERUSER_COMMAND, "pacman-key", "--populate", "archlinux"], check=True)
-
-            if cli_continue_question(message, config.EXTRA_INFO):
-                install_packages(config.PACKAGES)
-
-                if config.OS_NAME == "Steam":
-                    subprocess.run([config.SUPERUSER_COMMAND, "sed", '-i', 's/mymachines resolve/mymachines mdns_minimal [NOTFOUND=return] resolve/', '/etc/nsswitch.conf'], check=True)
-                    subprocess.run([config.SUPERUSER_COMMAND, "locale-gen"], check=True)
-                    subprocess.run([config.SUPERUSER_COMMAND, "systemctl", "enable", "--now", "avahi-daemon"], check=True)
-                    subprocess.run([config.SUPERUSER_COMMAND, "systemctl", "enable", "--now", "cups"], check=True)
-                    subprocess.run([config.SUPERUSER_COMMAND, "steamos-readonly", "enable"], check=True)
-        else:
-            logos_error(f"The script could not determine your {config.OS_NAME} install's package manager or it is unsupported. Your computer is missing the command(s) {missing_cmd}. Please install your distro's package(s) associated with {missing_cmd} for {config.OS_NAME}.\n{config.EXTRA_INFO}")
+def steam_preinstall_dependencies():
+        subprocess.run([config.SUPERUSER_COMMAND, "steamos-readonly", "disable"], check=True)
+        subprocess.run([config.SUPERUSER_COMMAND, "pacman-key", "--init"], check=True)
+        subprocess.run([config.SUPERUSER_COMMAND, "pacman-key", "--populate", "archlinux"], check=True)
+
+def steam_postinstall_dependencies():
+    subprocess.run([config.SUPERUSER_COMMAND, "sed", '-i', 's/mymachines resolve/mymachines mdns_minimal [NOTFOUND=return] resolve/', '/etc/nsswitch.conf'], check=True)
+    subprocess.run([config.SUPERUSER_COMMAND, "locale-gen"], check=True)
+    subprocess.run([config.SUPERUSER_COMMAND, "systemctl", "enable", "--now", "avahi-daemon"], check=True)
+    subprocess.run([config.SUPERUSER_COMMAND, "systemctl", "enable", "--now", "cups"], check=True)
+    subprocess.run([config.SUPERUSER_COMMAND, "steamos-readonly", "enable"], check=True)
```

Comment:
We could make this more generic:

Have a preinstall() and postinstall(), then check the OS before running the associated commands.

Reply:
This looks fine for now, unless we find out that other systems have similar kinds of pre/post-install tasks.